### PR TITLE
Added operators for Java::Lang::String and Array

### PR DESCRIPTION
**Pull request by ghost**
- [x] build operator [] (set, get) for java.util.Arrays
   Example: 
   Array a = {1, 2, 4};
   a[2] = 3;
   assertEquals(3, a[2]);

- [x] build operator [] (set, get) java.lang.String
   Example:
   String a = "123";
   a[2] = 3;
  assertEquals('3', a[2]);

- [x] Reformat code Arrays.hpp, Object.hpp, String.cpp, String.hpp, StringTest.cpp
- [x] Refactor code Arrays.hpp using macro

### CODE REVIEW

File: java/lang/String/StringTest.cpp [hunk 1]
```diff
@@ -307,7 +307,7 @@ TEST (JavaLang, StringCopyValueOf) {
     }
 }
 
-    // FIXME
+// FIXME
```
Comment (dthongvl) [inline java/lang/String/StringTest.cpp:310]: Please fix this or describe why you cant fix.

File: java/util/Arrays/Arrays.hpp [hunk 2]
```diff
@@ -24,727 +24,556 @@
...
+            COPY_OF(float);
+            COPY_OF(double);
+
+            template<typename T>
+            static Array<T> copyOf(Array<T> original, int newLength) {
+                return copyOf0(original, newLength, NULL);
+            }
+
+            template<typename T, typename U>
+            static Array<T> copyOf(Array<U> u, int newLength, T newType) {
```
Comment (dthongvl) [inline java/util/Arrays/Arrays.hpp]: Please rename u

File: java/util/Arrays/Arrays.hpp [hunk 3]
```diff
@@ -24,727 +24,556 @@
...
+                    if (original.get(index) != target.get(index)) {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            template<typename T>
+            static void fill0(Array<T> *a, int fromIndex, int toIndex, T val) {
```
Comment (dthongvl) [inline java/util/Arrays/Arrays.hpp]: Please rename variable

**Review by dthongvl — CHANGES_REQUESTED** [review]
Please rename variable to meaningful name.

File: java/lang/String/StringTest.cpp (hunk 1 above)
Comment (ghost) [reply java/lang/String/StringTest.cpp:310]: Yep, I just reformat code style, maybe you should ask the author of that method first.

File: java/util/Arrays/Arrays.hpp (hunk 3 above)
Comment (ghost) [reply java/util/Arrays/Arrays.hpp]: Yep, fixed.

**Pull request merged**